ccmbioinfo/lfs
# CARDS-1374: Convert SignIn (loginForm.js) component to a two-step process - one screen for username entry, one screen for password entry

This PR builds on top of the CARDS-1359 branch and provides the necessary UI and reverse proxy changes so that a user may login to CARDS either as a native Sling user or through a SAML Identity Provider (IdP). If a SAML IdP is configured, a two-step login process will be presented to the user where, the first step involves entering a _username_. If the _username_ is a native Sling user, the second step will be a password prompt for the given username. If the _username_ indicates a SAML login from a configured IdP (eg. `myuser@keycloakdemo`) the user will be redirected to login at the IdP login screen.

To test:

- Build this (`CARDS-1374`) branch.
- Follow the instructions in `Utilities/Administration/SAML/SAML_SETUP.md` to setup a local Keycloak Docker container as an IdP.
- Ensure that, when visiting `http://localhost:9090/`, you are able to login both as a native Apache Sling user (`admin`:`admin`) and as the `myuser@keycloakdemo` SAML user.
- After logging out as the `admin` user, ensure that no subsequent requests to `http://localhost:9090` contain the `sling.formauth` cookie.
- After logging out as the `myuser@keycloakdemo` SAML user, log back in as `myuser@keycloakdemo`. If the Single Sign On (SSO) session has not expired, you will not be prompted by Keycloak for the `myuser@keycloakdemo` _username_ and _password_ but you should still briefly see the Keycloak authentication redirection screen.
- Ensure that `fetchWithReLogin` (popup login dialog) functionality works properly for both a native Apache Sling user as well as a SAML authenticated user.
- Ensure that the _Security Tests_ in `Utilities/Administration/SAML/SAML_SETUP.md` also pass.
- Ensure that if `start_cards.sh` is invoked without the `--keycloak_demo` flag, that the regular one-step login form is presented to the user.
